### PR TITLE
build: refactor BLS library/bindings integration 

### PR DESCRIPTION
**Pull request by xdustinface**
Drops all unnecessary occurrences of `CGO_XXX` flags/variables in the repo, adds `dashpay/bls-signatures` as submodule (Makes most sense IMO but im open for other suggestions here) and updates the makefile to work with a new build script: `third_party/bls-signatures/bls-signatures.sh`.

### CODE REVIEW

**Comment by QuantumExplorer** [comment]
UtAck

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -13,16 +13,6 @@ jobs:
...
-      - name: Set GOPATH
-        run: |
-          echo "::set-env name=GOPATH::$(go env GOPATH)"
-      - name: Set CGO_LDFLAGS
-        run: |
-          echo "::set-env name=CGO_LDFLAGS::"-L${GOPATH}/src/github.com/dashpay/bls-signatures/build""
-      - name: Set CGO_CXXFLAGS
-        run: |
-          echo "::set-env name=CGO_CXXFLAGS::"-I${GOPATH}/src/github.com/dashpay/bls-signatures/src -I${GOPATH}/src/github.com/dashpay/bls-signatures/contrib/relic/include -I${GOPATH}/src/github.com/dashpay/bls-signatures/build/contrib/relic/include""
-      - name: test & coverage report creation
```
Comment (shumkov) [inline .github/workflows/lint.yaml:25]: Don't we need to keep this one?

**Review by shumkov — APPROVED** [review]
LGTM! 👍

File: third_party/bls-signatures/install.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+SCRIPT_PATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+BUILD_PATH="$SCRIPT_PATH/build"
+
+if [ ! -d $BUILD_PATH ]; then
+	echo "$BUILD_PATH doesn't exist. Run \"make build-bls\" first." >/dev/stderr
+	exit 1
+fi
+
+# Install the library
+sudo make -C $BUILD_PATH install
```
Comment (shumkov) [inline third_party/bls-signatures/install.sh:12]: Shall we move `sudo` to the tenderdash make file to make it more visible?

Comment (xdustinface) [reply third_party/bls-signatures/install.sh]: f7f7361, why not :)

**Review by shumkov — APPROVED** [review]
Looks good! 👍

**Pull request merged**